lib/engine/game/base.rb

```diff
@@ -592,8 +599,32 @@ def initialize_actions(actions)
         @loading = false
       end
 
+      def process_derived_action(action)
+        action = action_from_h(action) if action.is_a?(Hash)
+        action.derived = true
+
+        @action_queue.prepend(action)
+      end
+
       def process_action(action)
         action = action_from_h(action) if action.is_a?(Hash)
+        raise GameError, "Derived actions may not use process_action, especially #{action}" if action.derived
```

Comment:
This is probably (and hopefully) my most useless comment ever on this repo: instead of "especially" it should be "specially" or "specifically".